src/lib/sensor_calibration/Magnetometer.hpp

```diff
@@ -98,15 +97,21 @@ class Magnetometer
 
 	void Reset();
 
+	void SensorCorrectionsUpdate(bool force = false);
+
 	void UpdatePower(float power) { _power = power; }
 
 private:
+	uORB::Subscription _sensor_correction_sub{ORB_ID(sensor_correction)};
+
 	Rotation _rotation_enum{ROTATION_NONE};
 
 	matrix::Dcmf _rotation;
 	matrix::Vector3f _offset;
 	matrix::Matrix3f _scale;
+	matrix::Vector3f _thermal_offset;
```

Comment:
Make sure this field is reset with the rest. Maybe try side by side comparing Magnetometer.cpp and Accelerometer.cpp to keep them mostly in sync.

Reply:
Thanks.  That should be taken care of in SensorCorrectionsUpdate() just like the Accelerometer class and is now being called in VehicleMagnetometer per your review comment.  Thank you!